server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -568,34 +570,57 @@ private ExtensionsSettings readFromExtensionsYml(Path filePath) throws IOExcepti
             List<HashMap<String, ?>> unreadExtensions = new ArrayList<>((Collection<HashMap<String, ?>>) obj.get("extensions"));
             List<Extension> readExtensions = new ArrayList<Extension>();
             for (HashMap<String, ?> extensionMap : unreadExtensions) {
-                // Parse extension dependencies
-                List<ExtensionDependency> extensionDependencyList = new ArrayList<ExtensionDependency>();
-                if (extensionMap.get("dependencies") != null) {
-                    List<HashMap<String, ?>> extensionDependencies = new ArrayList<>(
-                        (Collection<HashMap<String, ?>>) extensionMap.get("dependencies")
-                    );
-                    for (HashMap<String, ?> dependency : extensionDependencies) {
-                        extensionDependencyList.add(
-                            new ExtensionDependency(
-                                dependency.get("uniqueId").toString(),
-                                Version.fromString(dependency.get("version").toString())
-                            )
+                try {
+                    // iterating the map to see whether any required field is missing
+                    String[] requiredFields = {
+                        "name",
+                        "uniqueId",
+                        "hostAddress",
+                        "port",
+                        "version",
+                        "opensearchVersion",
+                        "minimumCompatibleVersion" };
+                    for (String requireField : requiredFields) {
+                        if (!extensionMap.containsKey(requireField)) {
+                            throw new ParsingException(
+                                new XContentLocation(584, 10),
```

Comment:
This seems to denote the exact location of the error which wouldn't be a constant. But if you just switch to IOException you don't need this. :)

Reply:
changed to IOException